src/service/engine/index.ts

```diff
@@ -149,20 +165,35 @@ export class EngineService {
         ? this.CREATE_PARAMETER_NAMES_V2
         : this.CREATE_PARAMETER_NAMES;
 
-    if (Object.values(createOptions).some(v => v !== undefined)) {
+    const internalOptions = this.getInternalOptions();
+
+    const filteredCreateOptions = Object.entries(createOptions).filter(
+      ([, value]) => value !== undefined
+    );
+
+    if (filteredCreateOptions.length > 0 || internalOptions) {
       query += " WITH ";
-      for (const [key, value] of Object.entries(createOptions)) {
-        if (key in createParameterNames) {
-          if (key == "spec" && accountVersion >= 2) {
-            // spec value is provided raw without quotes for accounts v2
-            query += `${createParameterNames[key]} = ${value} `;
-          } else {
-            query += `${createParameterNames[key]} = ?`;
-            queryParameters.push(value);
-          }
+    }
+
+    for (const [key, value] of Object.entries(createOptions)) {
```

Comment:
Should we use `filteredCreateOptions` here instead of `createOptions`?

Reply:
Also, can we just iterate here on `filteredCreateOptions` + `internalOptions`?

Reply:
> can we just iterate here on filteredCreateOptions + internalOptions

Unfortunately this becomes complicated as we're doing a verification on `createParameterNames` before adding the parameter. When iterating we'll need to know if parameter is create option or internal to either go through map or not. This overcomplicates the logic within the loop so I think the most readable way of doing this is to have it separate.